### ISSUE TITLE
Castle of Illusion (227600)

### ISSUE THREAD

**Issue by KGOrphanides** [ValveSoftware/Proton#1498]
# Whitelist Request
- Name of the game to be whitelisted: Castle of Illusion
- Steam AppID of the game: 227600

## System Information
- GPU: AMD Radeon HD 6990
- Driver/LLVM version: Mesa 18.0.5
- Kernel version: 4.15.0-33-generic
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/KGOrphanides/c0fe99f0d9824a8ab434907637c07400
- Proton version: 3.7-3

## I confirm:
- [x] that pressing the `Play` button in the Steam client is sufficient.
- [ ] that [runtime config options](https://github.com/ValveSoftware/Proton#runtime-config-options)
      are necessary to run the game.
- [x] that no workarounds other than the mentioned ones are necessary.

## Issues
- [x] I haven't experienced any issues.
- [ ] There are no issues left open for this game.
- [ ] Although I consider the gaming experience equal to Windows there are
      remaining issues:

**Comment by lucifertdark** [comment]
Whitelisting this would be a mistake, I get nothing but a black screen, none of the workarounds I've found suggested elsewhere work at all.

[steam-227600.log](https://github.com/ValveSoftware/Proton/files/2391320/steam-227600.log)

https://gist.github.com/lucifertdark/584c14275612e8e3c4a8070d428d8a02

**Comment by elgabo86** [comment]
Yeah it's a black screen for Nvidia (396.54.05) but work with Intel HD Graphics 515

**Comment by lucifertdark** [comment]
> Yeah it's a black screen for Nvidia (396.54.05) but work with Intel HD Graphics 515

Ah so it's an Nvidia problem? that's good to know. And that's why it shouldn't be Whitelisted, it doesn't work for everyone out of the box.

**Comment by elgabo86** [comment]
Yes i can confirm that the black screen is only on Nvidia. 

And i can confirm too that fullscreen mode doesn't work and closes the game. (On Intel HD Graphics 515)

I'm on Archlinux 64bit, daily updated with last proton update.
(Sorry for my poor english)

**Comment by lucifertdark** [comment]
Drivers updated to 396.54.09, Proton 3.16-3 Beta, the game is dead. Time to start a list of games that will never work?

**Comment by kode54** [comment]
It's not working with my RX 480, so it's not really proper outside the Nvidia ecosystem, either.

https://gist.github.com/kode54/3ff952f39b407d6f806a6779838f5335#file-system_stats_rx480-txt

Garbage textures everywhere. I had to guess what the menus were to exit the game.

E: I've switched to Arch, and I'm using Mesa 18.3.1, and it now runs perfectly. System stats updated accordingly.

E2: Spoke too soon. Graphics glitches like before, but only when I quit from the game back to the title screen. Then once the menu appeared again, they were gone.

**Comment by RiantAnda** [comment]
This game finally worked for me with Proton 3.16-6 Beta and the new "PROTON_FORCE_LARGE_ADDRESS_AWARE=1" flag.
The game started up on the left upper part of the screen, but after going to options and changing the display resolution, fullscreen(1440p) with native display resolution was possible.
Framerate is a bit low in certain areas of the game...

**Comment by lucifertdark** [comment]
I'm impressed, I could never get this game to work even in Windows, now with the new flag & 3.16-6 Beta the game runs like a charm. The game even picked up the resolution so I didn't have to change anything.

I'm now using the 415.25 drivers from Nvidia, the game is working as it was intended. no sound or graphical issues whatsoever.

**Comment by ghost** [comment]
Proton:3.16-6
Distro:Debian GNU/Linux 9.7 (stretch) (64 bits)
Kernel:4.20.5
RAM:6 GB
GPU Driver:NVIDIA 415.22.05
GPU:NVIDIA GeForce GTX 750
CPU:AMD Phenom II X6 1090T

thanks a lot it work like a charm with PROTON_FORCE_LARGE_ADDRESS_AWARE=1 %command%

**Comment by RiantAnda** [comment]
The `PROTON_FORCE_LARGE_ADDRESS_AWARE=1 %command%` seems superfluous. The game now runs without any workarounds on Nvidia hardware.

@kisak-valve and @KGOrphanides 
I think the "Whitelist Request" label can be reinstalled again!

This game also works fine with D9VK, e.g using `PROTON_USE_D9VK=1 %command%`.
Seems to be running more smooth with it.

**Comment by kisak-valve** [comment]
Is anyone seeing an issue with this game that would block it from being whitelisted?

**Comment by lucifertdark** [comment]
Fully working with no launch commands or messing about with Winetricks needed on Proton 4.11-3, Whitelist away unless someone else finds a bug.
Nvidia drivers I'm using are 430.40.
